AdyenActions/Components/Redirect/RedirectComponent.swift

```diff
@@ -130,13 +155,45 @@ public final class RedirectComponent: ActionComponent {
     private func registerRedirectBounceBackListener(_ action: RedirectAction) {
         RedirectListener.registerForURL { [weak self] returnURL in
             guard let self = self else { return }
-            
-            let additionalDetails = RedirectDetails(returnURL: returnURL)
-            let actionData = ActionComponentData(details: additionalDetails, paymentData: action.paymentData)
-            self.delegate?.didProvide(actionData, from: self)
+            self.didOpen(url: returnURL, action)
         }
     }
     
+    private func didOpen(url returnURL: URL, _ action: RedirectAction) {
+        if let redirectStateData = action.nativeRedirectData {
+            handleNativeMobileRedirect(withReturnURL: returnURL, redirectStateData: redirectStateData, action)
+        } else {
+            do {
+                callDidProvide(redirectDetails: try RedirectDetails(returnURL: returnURL), action)
+            } catch {
+                delegate?.didFail(with: error, from: self)
+            }
+        }
+    }
+    
+    private func handleNativeMobileRedirect(withReturnURL returnURL: URL, redirectStateData: String, _ action: RedirectAction) {
+        guard let queryString = returnURL.query else {
+            delegate?.didFail(with: Error.invalidRedirectParameters, from: self)
+            return
+        }
+        let request = NativeRedirectResultRequest(redirectData: redirectStateData,
+                                                  returnQueryString: queryString)
+        apiClient.perform(request) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case let .failure(error):
+                self.delegate?.didFail(with: error, from: self)
+            case let .success(response):
+                self.callDidProvide(redirectDetails: response, action)
+            }
+        }
+    }
+
+    private func callDidProvide(redirectDetails: RedirectDetails, _ action: RedirectAction) {
```

Comment:
Instead of the word `call` how about `notifyDelegateForDidProvide` or something of this sort?